### ISSUE TITLE
[OKD4.7] Web-UI broken on "Create PersistentVolumeClaim"

### ISSUE THREAD

**Issue by devzeronull** [openshift/console#8452]
Hi,

we experienced an error on the Web-UI in 4.7.0-0.okd-2021-03-21-094146 Release (bare-metal UPI).
When we click in Administrator View on "PersistentVolumeClaims" -> "Create PersistentVolumeClaim" (URL: https://console-openshift-console.apps.clustername.basedomain/k8s/ns/default/persistentvolumeclaims/~new/form") 
![grafik](https://user-images.githubusercontent.com/77630137/112304265-93a8cf00-8c9d-11eb-9a39-5649c8183751.png)

the page gets loaded and after a  short flickering showing the Forms content for a millisecond it stays completely empty - like being overlayed with a white page or the page content blended out:
![grafik](https://user-images.githubusercontent.com/77630137/112304481-d23e8980-8c9d-11eb-88e7-48dc5b2b8b4f.png)

In 4.6 it would look like this:
![grafik](https://user-images.githubusercontent.com/77630137/112304666-0ade6300-8c9e-11eb-85c9-b0f1e9ffe446.png)

When looking at the pages source code it seems to be loaded having the following JavaScript-error:
![grafik](https://user-images.githubusercontent.com/77630137/112304828-3e20f200-8c9e-11eb-964d-180bef43192a.png)

> **TypeError: can't convert undefined to object**
>     d create-pvc-chunk-e6298432dd020e2cd449.min.js:1
>     k create-pvc-chunk-e6298432dd020e2cd449.min.js:1
>     Ia vendors~main-chunk-3652a4e54bce917198bc.min.js:166203
>     Ws vendors~main-chunk-3652a4e54bce917198bc.min.js:166203
>     unstable_runWithPriority vendors~main-chunk-3652a4e54bce917198bc.min.js:166497
>     li vendors~main-chunk-3652a4e54bce917198bc.min.js:166203
>     Ws vendors~main-chunk-3652a4e54bce917198bc.min.js:166203
>     Fs vendors~main-chunk-3652a4e54bce917198bc.min.js:166203
>     Ps vendors~main-chunk-3652a4e54bce917198bc.min.js:166203
>     hi vendors~main-chunk-3652a4e54bce917198bc.min.js:166203
>     unstable_runWithPriority vendors~main-chunk-3652a4e54bce917198bc.min.js:166497
>     li vendors~main-chunk-3652a4e54bce917198bc.min.js:166203
>     hi vendors~main-chunk-3652a4e54bce917198bc.min.js:166203
>     di vendors~main-chunk-3652a4e54bce917198bc.min.js:166203
>     Is vendors~main-chunk-3652a4e54bce917198bc.min.js:166203
>     notify vendors~main-chunk-3652a4e54bce917198bc.min.js:39200
>     notifyNestedSubs vendors~main-chunk-3652a4e54bce917198bc.min.js:39200
>     notifySubscribers vendors~main-chunk-3652a4e54bce917198bc.min.js:109462
>     handleChangeWrapper vendors~main-chunk-3652a4e54bce917198bc.min.js:39200
>     v vendors~main-chunk-3652a4e54bce917198bc.min.js:24762
>     b main-chunk-0e0280b47986f6ccc022.min.js:1
>     dispatch vendors~main-chunk-3652a4e54bce917198bc.min.js:24762
>     s main-chunk-0e0280b47986f6ccc022.min.js:1
>     s main-chunk-0e0280b47986f6ccc022.min.js:1
>     i main-chunk-0e0280b47986f6ccc022.min.js:1
> vendors~main-chunk-3652a4e54bce917198bc.min.js:166203
> **stopped watching storage.k8s.io~v1~StorageClass before finishing incremental loading with error TypeError: can't convert undefined to object! main-chunk-0e0280b47986f6ccc022.min.js:1**

It is reproduceable in 4.7, but not in 4.6 and seems like it could be related to the templating/branding issue on the first 4.7 Release...

Can you confirm that?

Best regards

**Comment by klzsysy** [comment]
me too

**Comment by spadgett** [comment]
Hi, I'm not able to reproduce. It's likely something in the data on your particular cluster, most likely something about one of the StorageClass resources that console doesn't like. Are you able to share the output of `oc get storageclasses -o yaml`?

/kind bug

cc @zherman0 @cloudbehl

**Comment by larueli** [comment]
We experience the same issue with `4.7.0-0.okd-2021-04-24-103438`.

We have two storage classes :

<details>
<summary>`oc get storageclasses -o yaml`</summary>

```
apiVersion: v1
items:
- allowVolumeExpansion: true
  apiVersion: storage.k8s.io/v1
  kind: StorageClass
  metadata:
    creationTimestamp: "2021-02-25T20:33:41Z"
    managedFields:
    - apiVersion: storage.k8s.io/v1
      fieldsType: FieldsV1
      fieldsV1:
        f:allowVolumeExpansion: {}
        f:parameters:
          .: {}
          f:clusterID: {}
          f:csi.storage.k8s.io/fstype: {}
          f:csi.storage.k8s.io/node-stage-secret-name: {}
          f:csi.storage.k8s.io/node-stage-secret-namespace: {}
          f:csi.storage.k8s.io/provisioner-secret-name: {}
          f:csi.storage.k8s.io/provisioner-secret-namespace: {}
          f:imageFeatures: {}
          f:imageFormat: {}
          f:pool: {}
        f:provisioner: {}
        f:reclaimPolicy: {}
        f:volumeBindingMode: {}
      manager: Mozilla
      operation: Update
      time: "2021-02-25T20:33:41Z"
    name: ceph
    resourceVersion: "53739199"
    selfLink: /apis/storage.k8s.io/v1/storageclasses/ceph
    uid: cebfcb6a-1df5-45ac-83e2-4c5ccea3279f
  parameters:
    clusterID: rook-ceph-external
    csi.storage.k8s.io/fstype: ext4
    csi.storage.k8s.io/node-stage-secret-name: rook-csi-rbd-node
    csi.storage.k8s.io/node-stage-secret-namespace: rook-ceph-external
    csi.storage.k8s.io/provisioner-secret-name: rook-csi-rbd-provisioner
    csi.storage.k8s.io/provisioner-secret-namespace: rook-ceph-external
    imageFeatures: layering
    imageFormat: “2”
    pool: openshift
  provisioner: rook-ceph.rbd.csi.ceph.com
  reclaimPolicy: Delete
  volumeBindingMode: Immediate
- apiVersion: storage.k8s.io/v1
  kind: StorageClass
  metadata:
    annotations:
      storageclass.kubernetes.io/is-default-class: "false"
    creationTimestamp: "2020-11-13T00:23:29Z"
    managedFields:
    - apiVersion: storage.k8s.io/v1
      fieldsType: FieldsV1
      fieldsV1:
        f:parameters:
          .: {}
          f:archiveOnDelete: {}
        f:provisioner: {}
        f:reclaimPolicy: {}
        f:volumeBindingMode: {}
      manager: oc
      operation: Update
      time: "2020-11-13T00:23:29Z"
    - apiVersion: storage.k8s.io/v1
      fieldsType: FieldsV1
      fieldsV1:
        f:metadata:
          f:annotations:
            .: {}
            f:storageclass.kubernetes.io/is-default-class: {}
      manager: Mozilla
      operation: Update
      time: "2021-01-20T13:06:42Z"
    name: nfs-proxmox-vm
    resourceVersion: "35969163"
    selfLink: /apis/storage.k8s.io/v1/storageclasses/nfs-proxmox-vm
    uid: bbe2124c-6740-41d8-b7a9-a900437cc2a9
  parameters:
    archiveOnDelete: "true"
  provisioner: nfs-proxmox-vm
  reclaimPolicy: Delete
  volumeBindingMode: Immediate
kind: List
metadata:
  resourceVersion: ""
  selfLink: ""
```

</details>

We experience this issue only on the ceph storage class. When I go on the page to create the pvc, it works, but when I select ceph, it renders a blank page.

Here is the error shown in the console output, hope it help ?

![image](https://user-images.githubusercontent.com/37900449/116077532-e1966580-a695-11eb-912c-8ff7ab92f2ea.png)

Feel free to ask more details !

**Comment by spadgett** [comment]
It looks like this was introduced in 4.7.z by #8234

/assign @GowthamShanmugam
cc @bipuladh

**Comment by GowthamShanmugam** [comment]
> It looks like this was introduced in 4.7.z by #8234
> 
> /assign @GowthamShanmugam
> cc @bipuladh

Fixing this issue

**Comment by GowthamShanmugam** [comment]
> > It looks like this was introduced in 4.7.z by #8234
> > /assign @GowthamShanmugam
> > cc @bipuladh
> 
> Fixing this issue

Fix PR: https://github.com/openshift/console/pull/8784, removed wrong commit(accidentally modified few files in rebase, reverted back those changes)

![pvc_page_blank](https://user-images.githubusercontent.com/15073590/116221943-b9793600-a76b-11eb-95b5-53bb0340ede1.png)

**Comment by spadgett** [comment]
Fixed by #8452.

/close
